ansible/roles/redpanda_broker/tasks/prepare-data-dir.yml

```diff
@@ -0,0 +1,86 @@
+---
+- name: Set device_info fact
+  ansible.builtin.set_fact:
+    device_info: "{{ hostvars[inventory_hostname].ansible_devices }}"
+    nvme_device_ids: "{{ hostvars[inventory_hostname].ansible_devices.keys() | map('regex_search', 'nvme.*') | select('string') | list }}"
+  tags:
+    - prep_data_dir
+
+- name: Set nvme_devices_for_raid fact
+  ansible.builtin.set_fact:
+    nvme_devices_for_raid: '{{ (nvme_devices_for_raid | default([])) + ["/dev/" + item] }}'
+  loop: "{{ nvme_device_ids }}"
+  when: device_info[item]["partitions"] | length == 0
+  tags:
+    - prep_data_dir
+
+- name: Set nvme_devices_for_raid fact with default value
+  ansible.builtin.set_fact:
+    nvme_devices_for_raid: "{{ (nvme_devices_for_raid | default([])) }}"
+  tags:
+    - prep_data_dir
+
+- name: Set ephemeral_disk fact
+  ansible.builtin.set_fact:
+    ephemeral_disk: "{{ (ephemeral_disk | default(false)) }}"
+  tags:
+    - prep_data_dir
+
+- name: handle mdadm
+  tags:
+    - prep_data_dir
+  block:
+    - name: Define mdadm_arrays variable
+      ansible.builtin.set_fact:
+        mdadm_arrays:
+          - name: md0
+            devices: "{{ nvme_devices_for_raid }}"
+            community.general.filesystem: xfs
+            level: 0
+            mountpoint: /mnt/vectorized
+            state: present
+    - name: Run mdadm
+      ansible.builtin.include_role:
+        name: mrlesmithjr.mdadm
+      when: nvme_devices_for_raid|length > 1
+
+- name: mounting file system
+  tags:
+    - prep_data_dir
+  block:
+    - name: Create XFS file system
+      community.general.filesystem:
+        fstype: xfs
+        dev: "{{ nvme_devices_for_raid[0] }}"
+    - name: Mount NVMe device
+      ansible.builtin.mount:
+        path: /mnt/vectorized
+        src: "{{ nvme_devices_for_raid[0] }}"
+        fstype: xfs
+        opts: "{{ 'defaults,nofail' if ephemeral_disk else 'defaults'}}"
+        dump: "{{ 1 if ephemeral_disk else 0}}"
+        passno: "{{ 2 if ephemeral_disk else 0}}"
+        state: mounted
+      when: nvme_devices_for_raid|length == 1
+
+- name: Create data path
+  ansible.builtin.file:
+    path: /mnt/vectorized/redpanda
+    state: directory
+    owner: root
+    group: root
+    mode: ugo=rwx
```

Comment:
Doesn't this result in permissions that are mode 777?

Reply:
That's what it was set to do I'm loathe to change it even if I agree it isn't great

Reply:
See the linter PR for explicit mode setting

Reply:
if this is generating 777, this is a security hole.

Reply:
I don't have a good way to test the ramifications of a change here barring standard "can we produce and consume a message from the cluster". If you can recommend one I'd be much happier making a change. 

I don't think it will have an impact changing it to 644 or something, but I'm not able to verify it.

Reply:
it's a directory, so it'll need to be 755 at least.

Reply:
I guess another way to put it is that there were only three file/directory creation tasks that have a mode set and this 777 is two of them. I don't know enough yet to say whether that's because someone threw their hands up and yolo'd or because it is actually necessary. I don't think it'd be a problem changing it, but I don't know what edge cases I haven't thought of yet 😓

Reply:
so there's going to me two that I'm aware of:

the directory created where we overlay the filesystem for RP. This should definitely be 755 (and arguably, it should be immutable to prevent certain failure conditions from causing rootfs to fill -- but immutable is not important right now).

the second is the top level directory on the filesystem we create. This should be 755 as well since only redpanda shouldbe writing to it. Challenge is from an ordering perspective. you have to ensure the mount point directory gets created and chmod'd, then mount the fs, then re-do the operation. In practice, it sort of happens on subsequent runs because of the mount overlay.

Reply:
changed to 775 in this PR https://github.com/redpanda-data/deployment-automation/pull/159/commits/a0f508df18d8dc1bd576772dd2d70f303ed57ea3